scipy/sparse/linalg/_dsolve/tests/test_linsolve.py

```diff
@@ -365,9 +364,9 @@ def test_ndarray_support(self):
     def test_gssv_badinput(self):
         N = 10
         d = arange(N) + 1.0
-        A = spdiags((d, 2*d, d[::-1]), (-3, 0, 5), N, N)
+        A = dia_array(((d, 2*d, d[::-1]), (-3, 0, 5)), shape=(N, N))
 
-        for spmatrix in (csc_matrix, csr_matrix):
+        for spmatrix in (csc_array, csr_array):
```

Comment:
Should we rename spmatrix here to something less matrix-y? It's just a name, but it could be confusing.